core/tx_pool.go

```diff
@@ -1046,16 +1046,19 @@ func (pool *TxPool) runReorg(done chan struct{}, reset *txpoolResetRequest, dirt
 	// because of another transaction (e.g. higher gas price).
 	if reset != nil {
 		pool.demoteUnexecutables()
+
+		// Update all accounts to the latest known pending nonce
+		nonces := make(map[common.Address]uint64)
+		for addr, list := range pool.pending {
+			txs := list.Flatten() // Heavy but will be cached and is needed by the miner anyway
+			nonces[addr] = txs[len(txs)-1].Nonce()+1
+		}
+		pool.pendingNonces.resetAll(nonces)
 	}
 	// Ensure pool.queue and pool.pending sizes stay within the configured limits.
 	pool.truncatePending()
 	pool.truncateQueue()
 
-	// Update all accounts to the latest known pending nonce
```

Comment:
I don't think it's correct. `truncatePending` will drop the overflowed transactions from the pending list. In the original code, the pending nonces will be set back.

Reply:
1. But in "truncatePending", the "nonce" of the account will be updated through "pool.pendingNonces.setIfLower" while deleting the overflowed transaction. The account that needs to be updated "nonce" has been updated in this function, so There is no need to reset the account again outside the function.
2. In the original code, if you are performing the "reset" action, first execute "truncatePending". Since the value of "pool.pendingNonces" has been cleared in the "reset" function, the subsequent steps are executed. When truncatePending" deletes the overflow transaction, "pool.pendingNonces.setIfLower" actually reads the "nonce" in "StateDB" to judge, the condition is not established and will not be updated to the highest "nonce", and finally by direct update All accounts are realized
3. So only need to update all "nonce" to "pool.pendingNonces" at the time of "reset", and after the update is completed, then execute "truncatePending", this time if the function deletes the overflow transaction, "pool .pendingNonces.setIfLower" is able to update the "nonce" of the corresponding account